pkg/wal/processor/transformer/wal_postgres_transformer_parser.go

```diff
@@ -78,7 +79,11 @@ func (v *PostgresTransformerParser) ParseAndValidate(rules []TableRules) (map[st
 
 			// validate that the transformer is compatible with the column type
 			if !pgTypeCompatibleWithTransformerType(transformer.CompatibleTypes(), datatype) {
-				return nil, fmt.Errorf("transformer specified for column '%s' in table %s does not support pg data type with oid: %d", colName, tableKey, datatype)
+				typeForOid, ok := pgtype_map.TypeForOID(datatype)
```

Comment:
Since `TypeForOID` is not defined in this PR, a general question: what kind of permissions does `pgstream` need on source except logical replication to be able to read pg catalog? Is this documented somewhere?

Reply:
not sure if I understood you question correctly, but this method doesn't query postgres to get the type. It's just a library that has the mapping oid->type name